ext/posix/posix.c

```diff
@@ -477,7 +477,11 @@ PHP_FUNCTION(posix_ttyname)
 #if defined(ZTS) && defined(HAVE_TTYNAME_R) && defined(_SC_TTY_NAME_MAX)
 	buflen = sysconf(_SC_TTY_NAME_MAX);
 	if (buflen < 1) {
+#if defined(__FreeBSD__) && defined(_SC_PAGESIZE)
```

Comment:
what makes you think _SC_PAGESIZE might not be defined ?

Reply:
I can't recollect the exact reason at the moment but so far from my Goldfish memory I think it was that either Zend or zts redefines the page size with it's real size and checks in this similar fashion. Or it might have been too old when SC_PAGESIZE was not defined in FreeBSD at all. Because at one point there was only `_SC_PAGE_SIZE` rather then `_SC_PAGESIZE`

Reply:
I looked even at freebsd 5.0, seems [it was already there](https://github.com/freebsd/freebsd-src/blob/02bbf3743bb6bab2aa930ccbed9048456434bb62/include/unistd.h#L206).